src/Cardano/Wallet/Primitive/AddressDiscovery.hs

```diff
@@ -254,31 +259,97 @@ nextAddresses !key (AddressPoolGap !g) !cc !fromIx =
         (>= fromIx)
     newAddress = keyToAddress . deriveAddressPublicKey key cc
 
+{-------------------------------------------------------------------------------
+                            Pending Change Indexes
+-------------------------------------------------------------------------------}
+
+-- | An ordered set of pending indexes. This keep track of indexes used
+newtype PendingIxs = PendingIxs [Index 'Soft 'AddressK]
+    deriving stock (Generic, Show)
+instance NFData PendingIxs
+
+-- | An empty pending set of change indexes.
+--
+-- NOTE: We do not define a 'Monoid' instance here because there's no rational
+-- of combining two pending sets.
+emptyPendingIxs :: PendingIxs
+emptyPendingIxs = PendingIxs mempty
+
+-- | Update the set of pending indexes by discarding every indexes _below_ the
```

Comment:
@KtorZ Can I suggest calling it `discardIxsBelow` then?

Reply:
Hmmm.. I wouldn't. So that the _logic_ remains encapsulated in the `PendingIxs` structure. What the update does is actually irrelevant to the caller so I wouldn't have the name conveying any behavior here but rather, be fairly abstract like "update" or "adjust".